config/routes.rb

```diff
@@ -37,6 +37,7 @@
   match '/worker_health', to: "dashboards#build_history_by_worker", via: :get, as: :build_history_by_worker
 
   match 'builds/:id' => "builds#build_redirect", :via => :get, :as => :build_redirect
+  match 'builds/ref/:ref' => "builds#build_ref_redirect", :via => :get, :as => :build_ref_redirect
```

Comment:
It would be slick to use [Segment Constraints](http://guides.rubyonrails.org/routing.html#segment-constraints) here to make `/builds/:something` work for either a build ID or a Git ref.